tests/test_sdist.py

```diff
@@ -0,0 +1,47 @@
+import subprocess
```

Comment:
Flit has the annoying "feature" that `flit build --sdist` and `pipx run build --sdist` (or any other PEP 517 builder) produce different results. When using `flit`, it uses Git to decide what goes in the SDist. When using standard tools, it does not. This test module compares the SDists produced the two ways to make sure standard tooling can create a valid SDist.

There's plans to add a feature directly to Flit to help with this - once that is in, this can reduce to just a simple check, probably something like `flit validate-sdist --check` or something.

Reply:
The thing that was bugging me was that you specified this test to be run as a specific `tox` environment, but on GitHub Actions you run it on all Python versions rather than in a separate job. Does this need to be run separately on each Python version?

Reply:
No, I set the tox environment specifically because it only needs to run once. But then the CI didn't use tox, so I just added the deps there.  I can easily set the deps to only be added in one matrix run, or add a custom job for it in CI if you'd prefer. The check doesn't take very long, but it's not instant either.

Reply:
Wouldn't separate job most likely just slow down CI?

Reply:
> Wouldn't separate job most likely just slow down CI?

No since it would run parallel to the test suites, or did you mean something else?

Reply:
If the flit PEP 517 back-end doesn't use git to determine the manifest contents for the sdist, what does it use then?

Reply:
Basically a MANIFEST.in: https://github.com/henryiii/wheel/blob/95dce90f5c0b0ff9cec1f135dcff0a2d1abc82dc/pyproject.toml#L44-L87 - that's what this job checks, it makes sure this specification stays in sync with what it would have selected from git.

Reply:
I was just thinking, if setting up the test environment takes more time than running the test, separate job is not time-efficient.

Reply:
I've limited this check to one version of Python only in CI now.

Reply:
Am I correct in understanding that the only reason we have this duality is so we can generate a stub `setup.py` in the sdist?

Reply:
Not really, this helps make sure that standards-based builds keep working. If you add a new file, it's easy to forget to add it to the correct include/exclude list. This check uses the git-aware Flit tool to see if flit-core is producing the same SDist (minus the generated setup.py) that a standards-based build would produce. It's the same sort of thing that `check-manifest` for Setuptools used to do.

If you add a new file in git, this check will start failing until that file is matched via the include or exclude lists.

This is entirely a stand-in for a proper tool in Flit that would do exactly this - see https://github.com/pypa/flit/issues/522#issuecomment-1409108752. Once that's implemented, this check could be replaced with that.